packages/nest/package.py

```diff
@@ -0,0 +1,179 @@
+
+    version('develop', git=url)
+
+    variant('mpi',      default=True,  description="Enable MPI support")
+    variant('openmp',   default=True,  description="Enable OpenMP support")
+    variant('python',   default=True,  description="Enable python bindings")
+    variant('gsl',      default=True,  description="Enable GNU Scientific Library")
+    variant('profile',  default=False, description="Enable profiling using Tau")
+    variant('knl',      default=False, description="Enable KNL specific flags")
+    variant('static',   default=False, description="Build static libraries")
+
+    depends_on('mpi', when='+mpi')
+    depends_on('gsl', when='+gsl')
+    depends_on('python@2.6:', when='+python')
+    depends_on('cmake@2.8.12:', type='build')
+    depends_on('tau', when='+profile')
+    depends_on('py-cython@0.19.2:', when='+python')
+
+    def profiling_wrapper_on(self):
+        if self.spec.satisfies('+profile'):
+            os.environ["USE_PROFILER_WRAPPER"] = "1"
+
+    def profiling_wrapper_off(self):
+        if self.spec.satisfies('+profile'):
+            del os.environ["USE_PROFILER_WRAPPER"]
+
+    def get_optimization_level(self):
+        flags = "-g"
+
+        if 'bgq' in self.spec.architecture:
+            flags += ' -O3'
+        else:
+            flags += ' -O2'
+
+        if self.spec.satisfies('+knl'):
+            flags = '-g -xmic-avx512 -O3 -qopt-report=5'
+
+        return flags
+
+    def install(self, spec, prefix):
+        build_dir = "spack-build-%s" % spec.version
+
+        with working_dir(build_dir, create=True):
+            c_compiler = spack_cc
+            cxx_compiler = spack_cxx
+
+            optflag = self.get_optimization_level()
+
+            if self.spec.satisfies('+profile'):
+                c_compiler = 'tau_cc'
+                cxx_compiler = 'tau_cxx'
+            # for bg-q, our cmake is not setup properly
+            elif 'bgq' in self.spec.architecture and self.spec.satisfies('+mpi'):
+                    c_compiler = self.spec['mpi'].mpicc
+                    cxx_compiler = self.spec['mpi'].mpicxx
+
+            cmake_options = ['-DCMAKE_INSTALL_PREFIX:PATH=%s' % prefix,
+                             '-DCMAKE_C_FLAGS=%s' % optflag,
+                             '-DCMAKE_CXX_FLAGS=%s' % optflag,
+                             '-DCMAKE_C_COMPILER=%s' % c_compiler,
+                             '-DCMAKE_CXX_COMPILER=%s' % cxx_compiler,
+                             '-Dwith-ltdl=OFF',
+                             '-Dwith-readline=OFF']
+
+            # not sure the -Dwith-optimize=flags option is required if we already set CMAKE_C_FLAGS above
+            # but just to be safe
+            semicolon_separated_optflag = optflag.translate(string.maketrans(' ', ';'))
+            cmake_options.extend(['-Dwith-optimize=' + semicolon_separated_optflag])
+
+            if self.spec.satisfies('~python'):
+                cmake_options.extend(['-Dwith-python=OFF'])
```

Comment:
typical convention is `if satisfies(+python) .. else ...` (never order, just for consistency) I may not have followed this when I created packages in the beginning :)

for all below options...